pkg/rules/adapters/ingest.go

```diff
@@ -0,0 +1,93 @@
+package adapters
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/prometheus/prometheus/model/exemplar"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/storage"
+
+	"github.com/timescale/promscale/pkg/pgmodel/cache"
+	"github.com/timescale/promscale/pkg/pgmodel/ingestor"
+	"github.com/timescale/promscale/pkg/pgmodel/metrics"
+	"github.com/timescale/promscale/pkg/pgmodel/model"
+	"github.com/timescale/promscale/pkg/prompb"
+)
+
+var samplesIngested = metrics.IngestorItems.With(map[string]string{"type": "metric", "kind": "sample", "subsystem": "ruling"})
+
+type ingestAdapter struct {
+	ingestor *ingestor.DBIngestor
+}
+
+func NewIngest(ingestor *ingestor.DBIngestor) *ingestAdapter {
+	return &ingestAdapter{ingestor}
+}
+
+type appenderAdapter struct {
+	// Map allows us to group similar samples into a single prompb.TimeSeries
```

Comment:
This would need some limits (Eg max number of samples?)? Otherwise we can blow up the memory here.

Reply:
No, this isn't an issue. Technically, the engine has a limit of 50M samples, but a single promql instance query should not return even that much.